src/renderers/webgl/WebGLMorphtargets.js

```diff
@@ -191,7 +191,7 @@ function WebGLMorphtargets( gl, capabilities, textures ) {
 			program.getUniforms().setValue( gl, 'morphTargetInfluences', objectInfluences );
 
 			program.getUniforms().setValue( gl, 'morphTargetsTexture', entry.texture, textures );
-			program.getUniforms().setValue( gl, 'morphTargetsTextureSize', entry.size );
+			program.getUniforms().setValue( gl, 'morphTargetsTextureSize', [ entry.size.x, entry.size.y ] );
```

Comment:
Good catch! However, I suggest to fix the issue in `setValueV2i()` like so: 
```js
function setValueV2i( gl, v ) {

	const cache = this.cache;

	if ( v.x !== undefined ) {

		if ( cache[ 0 ] !== v.x || cache[ 1 ] !== v.y ) {

			gl.uniform2iv( this.addr, v.x, v.y );

			cache[ 0 ] = v.x;
			cache[ 1 ] = v.y;

		}

	} else {

		if ( arraysEqual( cache, v ) ) return;

		gl.uniform2iv( this.addr, v );

		copyArray( cache, v );

	}

}
```
In this way, the method is similar to `setValueV2f()` and we don't create an array inline over and over again.

Reply:
I would also update `setValueV3i()` and `setValueV4i()`.

Reply:
I wanted to avoid adding an additional branch for every use of this function to fix the (seemingly) single bad user.  I'll do a perf check of this vs. the array and update the diff if there's no major delta.